include/zephyr/sys/util.h

```diff
@@ -783,6 +783,35 @@ static inline void mem_xor_128(uint8_t dst[16], const uint8_t src1[16], const ui
 	mem_xor_n(dst, src1, src2, 16);
 }
 
+/**
+ * @brief Returns the number of bits set in a value
+ *
+ * @param value The value to count number of bits set of
+ * @param len The number of octets in @p value
+ */
+static inline size_t zephyr_count_bits(const void *value, size_t len)
```

Comment:
Tough, this could be just `sys_count_bits()` use the well-known `sys_` prefix and be aligned with existing utils, for unknown reasons some people prefer to use the `zephyr_` prefix. But `zephyr_` is not really better, because it is also used by many other software projects, and it does not minimize the likelihood of collisions. Instead, we should use UUID version 5 as a prefix, like `7f756fa7_4e84_529e_b406_276069b9bcd0_`.

```suggestion
static inline size_t 7f756fa7_4e84_529e_b406_276069b9bcd0_count_bits(const void *const value, const size_t len)
{
```

Reply:
I get your point, and I don't really have a strong preference myself. I went with `zephyr_` as that seemed like the least controversial output from https://github.com/zephyrproject-rtos/zephyr/issues/64627

> but zephyr_ is not really better, because it is also used by many other software projects

Can you share some of these projects so we can consider them properly?

Reply:
> Instead, we should use UUID version 5 as a prefix, like `7f756fa7_4e84_529e_b406_276069b9bcd0_`.

The C language prohibits starting identifiers with a number 🫣

Reply:
We can use 30a23962-5d6b-4c76-87aa-cc799d50a6a7 and with "Zephyr RTOS" as namespace resulting in c496e475_323e_5ca0_8fa1_39f91557c0e0.

Reply:
> Can you share some of these projects so we can consider them properly?

There are some cloud and cryptobro projects. Just search for zephyr.